typhos/display.py

```diff
@@ -61,25 +163,33 @@ class TyphosDeviceDisplay(TyphosBase, TyphosDesignerMixin, _DisplayTypes):
     Q_ENUMS(_DisplayTypes)
     TemplateEnum = DisplayTypes  # For convenience
 
-    def __init__(self, parent=None, **kwargs):
+    def __init__(self, parent=None, *, embedded_templates=None,
+                 detailed_templates=None, engineering_templates=None,
+                 **kwargs):
         # Intialize background variable
         self._forced_template = ''
-        self._last_macros = dict()
+        self._macros = dict()
         self._main_widget = None
         self._display_type = DisplayTypes.detailed_screen
 
-        # Without a device set
-        self.templates = {
-            templ.name: os.path.join(ui_dir, templ.name + '.ui')
-            for templ in self.TemplateEnum
+        self.templates = {type_.name: [] for type_ in DisplayTypes}
+
+        instance_templates = {
+            'embedded_screen': embedded_templates or [],
+            'detailed_screen': detailed_templates or [],
+            'engineering_screen': engineering_templates or [],
         }
+        for view, path_list in instance_templates.items():
+            paths = [pathlib.Path(p).expanduser().resolve() for p in path_list]
```

Comment:
Maybe also expand env vars?

Reply:
You have unexpanded environment variables in `PYDM_DISPLAY_PATH`? I'd expect bash would handle that prior to running typhos.

Or are you thinking of macros in the path list?

Reply:
@hhslepicka noted in private this isn't an issue and worked locally